src/test-apps/TestResourceIdentifier.cpp

```diff
@@ -410,12 +410,12 @@ static void CheckStringSerDes(nlTestSuite * inSuite, void * inContext)
     ResourceIdentifier resource_unknown_type(0xC001, 0x18b4300000000001ULL);
     ResourceIdentifier resource_uninitialized;
 
-    const char * resource_str               = "DEVICE_18B4300000000001";
-    const char * resource2_str              = "USER_18B4300000000001";
-    const char * resource_self_str          = "RESERVED_DEVICE_SELF";
-    const char * resource_uninitialized_str = "RESERVED_NOT_SPECIFIED";
-    const char * resource_unknown_type_str  = "(C001)_18B4300000000001";
-    const char * resource_unknown_str       = "WIDGET_18B4300000000001";
+    const char resource_str[]               = "DEVICE_18B4300000000001";
+    const char resource2_str[]              = "USER_18B4300000000001";
+    const char resource_self_str[]          = "RESERVED_DEVICE_SELF";
+    const char resource_uninitialized_str[] = "RESERVED_NOT_SPECIFIED";
+    const char resource_unknown_type_str[]  = "(C001)_18B4300000000001";
+    const char resource_unknown_str[]       = "WIDGET_18B4300000000001";
```

Comment:
Maybe add a test for object id < 16 hex characters?

Reply:
Added tests.  I've also updated the parsing code, it was incorrect previously for device IDs shorter than 8 hex characters.